### PR TITLE
Add Slack notifictions for job failures

### PR DESCRIPTION
**Pull request by bradfordb-vmware**
[#173364325]

Authored-by: Bradford D. Boyle <bradfordb@vmware.com>

### CODE REVIEW

File: ci/concourse/pipelines/gpdb_opensource_release.yml [hunk 1]
```diff
@@ -208,6 +225,8 @@ jobs:
       file: gpdb_artifacts/bin_gpdb.tar.gz
 
 - name: compile_gpdb_centos7
+  on_failure:
+    <<: *slack-alert
```
Comment (50wu) [inline ci/concourse/pipelines/gpdb_opensource_release.yml:229]: I like this idea to use yaml anchors and references to reduce duplication, it's cool! 👍
One question is how to know which slack channel is sent to?

Comment (baotingfang) [reply ci/concourse/pipelines/gpdb_opensource_release.yml:229]: @50wu You can ref here: https://github.com/pivotal/gp-continuous-integration/pull/220/files#diff-0dec79d3a103aa287c87379e3e19319aR29

Comment (baotingfang) [reply ci/concourse/pipelines/gpdb_opensource_release.yml:229]: @50wu Maybe you are interested in this article. https://blog.betomorrow.com/yaml-tips-and-tricks-for-concourse-ci-35a3b577a239

File: ci/concourse/pipelines/gpdb_opensource_release.yml [hunk 2]
```diff
@@ -192,6 +204,11 @@ resources:
 
 jobs:
 - name: compile_gpdb_centos6
+  on_failure: &slack-alert
+    put: slack-alert
+    params:
+      # yamllint disable-line rule:line-length
+      text: Hey team, <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|greenplum-database-release/$BUILD_JOB_NAME> failed.
```
Comment (baotingfang) [inline ci/concourse/pipelines/gpdb_opensource_release.yml:211]: I like these Env variables! 😄   details: https://concourse-ci.org/implementing-resource-types.html#resource-metadata

File: ci/concourse/pipelines/gpdb_opensource_release.yml (hunk 1 above)
Comment (50wu) [reply ci/concourse/pipelines/gpdb_opensource_release.yml:229]: @baotingfang Thanks for sharing this good articles!

Comment (bradfordb-vmware) [reply ci/concourse/pipelines/gpdb_opensource_release.yml:229]: The (default) Slack channel is configured in the Slack app's incoming webhooks. I've updated the credentials with some notes for the name of Slack app and the channel messages will be sent to.

**Pull request merged**